src/plugins/contextMenu/contextMenu.js

```diff
@@ -236,6 +236,41 @@ class ContextMenu extends BasePlugin {
     this.commandExecutor.execute.apply(this.commandExecutor, params);
   }
 
+  changeItems(removeItemFunction, newItems) {
+
+    if (typeof removeItemFunction === 'function') {
+      arrayEach(Object.entries(this.itemsFactory.predefinedItems), (entry) => {
+        let key = entry[0];
+        let item = entry[1];
+        if (removeItemFunction(key, item)) {
+          delete this.itemsFactory.predefinedItems[key];
+        }
+      });
+    }
+
+    arrayEach(newItems || [], (item) => this.itemsFactory.predefinedItems[item.name] = item);
```

Comment:
the second parameter takes a list of new items to add to the context list. Note that removes happen first, in case any of the mew items would get removed by the remove function.